e2e-tests/cypress/tests/integration/channels/accessibility/accessibility_sidebar_spec.ts

```diff
@@ -165,6 +165,81 @@ describe('Verify Accessibility Support in Channel Sidebar Navigation', () => {
     });
 });
 
+describe('Accessibility tests for RHS getting focus after buttons actions', () => {
+    let testUser;
+
+    before(() => {
+        cy.apiInitSetup().then(({team, user}) => {
+            testUser = user;
+
+            // # Login as the test user and visit the town-square channel
+            cy.apiLogin(testUser);
+            cy.visit(`/${team.name}/channels/town-square`);
+            cy.get('#postListContent').should('be.visible');
+        });
+    });
+
+    beforeEach(() => {
+        // # Close the RHS
+        cy.uiCloseRHS();
+        cy.get('#sidebar-right').should('not.exist');
+    });
+
+    afterEach(() => {
+        // # Close the RHS
+        cy.uiCloseRHS();
+        cy.get('#sidebar-right').should('not.exist');
+    });
+
+    it('Focus should be on RHS when opening Recent Mentions', () => {
+        // # Click the Recent Mentions button
+        cy.findByRole('button', {name: /Recent mentions/i}).click();
+
+        // * Verify RHS is open
+        cy.get('#sidebar-right').should('be.visible');
+
+        // * Check that the RHS container is focused
+        cy.get('.sidebar-right-container').should('be.focused');
+    });
+
+    it('Focus should be on RHS when opening Saved Messages', () => {
+        // # Click the Saved Messages button
+        cy.findByRole('button', {name: /Saved messages/i}).click();
+
+        // * Verify RHS is open
+        cy.get('#sidebar-right').should('be.visible');
+
+        // * Check that the RHS container is focused
+        cy.get('.sidebar-right-container').should('be.focused');
+    });
+
+    it('Focus should be on RHS when opening Members', () => {
+        cy.get('#channelHeaderInfo').should('exist');
+
+        // # Click the Members button
+        cy.get('#member_rhs').
+            should('be.visible').
+            click();
+
+        // * Verify RHS is open
+        cy.get('#sidebar-right').should('be.visible');
+
+        // * Check that the RHS container is focused
+        cy.get('.sidebar-right-container').should('be.focused');
+    });
+
+    it('Focus should be on RHS when opening Channel files', () => {
```

Comment:
I know some plugins are used for testing some stuff in e2e. I wonder if it is worth it to check whether a plugin opening the RHS puts the focus there. But completely 0/5.

Reply:
we haven't cover that part yet, even the plugin buttons that open the RHS do not work on focus + enter. There is a ticket to work that part.